### ISSUE TITLE
linter: false positve in jest/expect-expect

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#12905]
> @camc314 Hi! I have the same issue, still in `1.11.0`. 
> 
> `eslint-plugin-jest(expect-expect): Test has no assertions`
> 
> ```
>     it('test', async () => {
>         const array = [1];
> 
>         for (const element of array) {
>             expect(element).toBe(1);
>         }
>     });
> ``` 

 _Originally posted by @kennidenni in [#12869](https://github.com/oxc-project/oxc/issues/12869#issuecomment-3167485628)_

**Comment by kennidenni** [comment]
It works in 1.9.0, but gives a false positive in 1.10.0 and 1.11.0
